python/ray/job_config.py

```diff
@@ -18,14 +19,24 @@ class JobConfig:
     def __init__(
             self,
             worker_env=None,
+            python_worker_setup_hook=None,
             num_java_workers_per_process=1,
             jvm_options=None,
             code_search_path=None,
     ):
         if worker_env is None:
             self.worker_env = dict()
         else:
+            if not isinstance(worker_env, dict):
+                raise TypeError("worker_env must be a Dict[str, str].")
             self.worker_env = worker_env
+
+        if python_worker_setup_hook is not None and not callable(
+                python_worker_setup_hook):
+            raise TypeError(
+                "python_worker_setup_hook must be a serializable function.")
```

Comment:
nit: `must be a callable`? Serializable function would require a serialization check I think

Reply:
(nvm didn't see this was wip)